File: Laravel/app/Services/Handlers/DBHandlers/DBBackupHandler.php
<?php
/**
 * Created by PhpStorm.
 * User: tsao
 * Date: 2017/11/25
 * Time: 下午 11:15
 */

namespace App\Services\Handlers\DBHandlers;


use App\Services\Configs\Candidate;
use DB;

/**
 * 備份檔案
 * Class DBBackupHandler
 * @author steve.tsao
 * @package App\Services\Handlers\DBHandlers
 */
class DBBackupHandler extends AbstractDBHandler
{
    /**
     * 備份檔案處理
     * @author steve.tsao
     * @param Candidate $candidate 檔案資訊
     * @param null|string $target 檔案內容
     * @return string 檔案內容
     */
    public function Perform(Candidate $candidate, $target): string
    {
        $sql = 'INSERT INTO Backups (Backup_filename, Backup_filedatetime, Backup_filesize, Backup_blob)'
            . '        VALUES (?, ?, ?, ?)';

        DB::insert($sql, [
            $this->ConvertFileName($candidate->Name),
            $candidate->FileDateTime,
            $candidate->Size,
            $target,
        ]);

        return $target;
    }
}

File: NetCore/ConsoleOop/ConsoleOop/Handlers/DbHandlers/DBBackupHandler.cs
using MyBackupCandidate;
using MySql.Data.MySqlClient;

namespace ConsoleOop.Handlers.DbHandlers
{
    /// <summary>
    /// 備份檔案
    /// </summary>
    class DBBackupHandler : AbstractDBHandler
    {
        /// <summary>
        /// 備份檔案處理
        /// </summary>
        /// <param name="candidate">檔案資訊</param>
        /// <param name="target">檔案內容</param>
        /// <returns>檔案內容</returns>
        public override byte[] Perform(Candidate candidate, byte[] target)
        {
            using (MySqlCommand cmd = new MySqlCommand("INSERT INTO Backups (Backup_filename, Backup_filedatetime, Backup_filesize, Backup_blob) VALUES (@filename, @filedatetime, @filesize, @blob)", this.conn))
            {
                cmd.Parameters.Add("@filename", MySqlDbType.VarChar).Value = candidate.Name;
                cmd.Parameters.Add("@filedatetime", MySqlDbType.DateTime).Value = candidate.FileDateTime;
                cmd.Parameters.Add("@filesize", MySqlDbType.Int64).Value = candidate.Size;
                cmd.Parameters.Add("@blob", MySqlDbType.LongBlob).Value = target;
                cmd.ExecuteNonQuery();
            }

            return target;
        }
    }
}
